packages/datadog-instrumentations/src/next.js

```diff
@@ -154,6 +155,11 @@ function finish (ctx, result, err) {
     errorChannel.publish(ctx)
   }
 
+  const maybeNextRequest = nodeNextRequestsToNextRequests.get(ctx.req)
+  if (maybeNextRequest) {
+    ctx = { nextRequest: maybeNextRequest, ...ctx }
```

Comment:
The `ctx` object must remain the same across calls and shouldn't be replaced here.

Reply:
Would changing it to `ctx.nextRequest = maybeNextRequest` be ok?

Reply:
Yes, that's basically the expected use of `ctx`. The idea is that having the same object everywhere allows events to share data.